src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -249,14 +259,26 @@ else if (msg.what == ENQUEUE_EVENTS) {
                     else if (msg.what == FLUSH_QUEUE) {
                         logAboutMessageToMixpanel("Flushing queue due to scheduled or forced flush");
                         updateFlushFrequency();
-                        sendAllData(mDbAdapter);
-                        mDecideChecker.runDecideChecks(getPoster());
+                        if (SystemClock.uptimeMillis() >= mRetryAfter) {
```

Comment:
I think this should be `SystemClock.elapsedRealtime` rather than `uptimeMillis`, since we want to send even if we were asleep when the time passed.